pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -538,6 +556,127 @@ public void deleteFailed(ManagedLedgerException exception, Object ctx) {
+                    }
+                    batchSizeAndHashPair = positionRemovedFromCursor
+                            ? consumer.getPendingAcks().removeAndReturn(ledgerId, entryId)
+                            : consumer.getPendingAcks().get(ledgerId, entryId);
+                    if (batchSizeAndHashPair != null) {
+                        // Continue find the owner
+                        owner = consumer;
+                        break;
+                    }
+                }
+            }
+            if (owner == null) {
+                // Since we can not get how many msgs that were attempted to ack, just plus 1 into the
+                // "attemptAckMsgs".
+                attemptAckMsgs++;
+                log.warn("[{}][{}]{}-{}-{} skipped to reduce un-ack-msgs for {}:{}, because could not find the"
+                                + " message's owner. consumer size: {}",
+                        topicName, subName,
+                        ackFrom == null ? "null" : ackFrom.cnx(),
+                        ackFrom == null ? "null" : ackFrom.consumerId(),
+                        ackFrom == null ? "null" : ackFrom.consumerName(),
+                        ledgerId, entryId, getConsumers().size());
+                continue;
+            }
+            // Calculate messages actually acked in batch.
+            int actualAcked = 0;
+            if (batchMessagesAckedCount == BATCH_MESSAGE_ACKED_AT_ONCE) {
+                // All messages in batch were acked at once.
+                actualAcked = Math.max(batchSizeAndHashPair.firstInt(), 1);
+            } else if (batchMessagesAckedCount == BATCH_MESSAGE_ACKED_FIRST_PART) {
+                // First part of batch message acked.
+                // Regarding this case, only consumer knows how many messages in batch were acked, because
+                // the cursor do not know how many messages in the batch, only "consumer.pendingAcks" knows.
+                long[] ackSetWords = AckSetStateUtil.getAckSetArrayOrNull(position);
+                if (ackSetWords != null) {
+                    BitSetRecyclable ackSet = BitSetRecyclable.create().resetWords(ackSetWords);
+                    actualAcked = Math.max(batchSizeAndHashPair.firstInt() - ackSet.cardinality(), 0);
+                    ackSet.recycle();
+                }
+            } else {
+                // Following part of batch message acked.
+                // Regarding this case, only cursor know how many messages in batch were acked, because
+                // "consumer.pendingAcks" does not know how many messages were acked count before, only "cursor"
+                // knows.
+                actualAcked = batchMessagesAckedCount;
+            }
+            attemptAckMsgs += actualAcked;
+            // Reduce un-acknowledged messages.
+            owner.addAndGetUnAckedMsgs(owner, -actualAcked);
+            owner.updateBlockedConsumerOnUnackedMsgs(owner);
+            if (log.isInfoEnabled()) {
```

Comment:
Is info level too high for this log message?

Reply:
Changed to `debug` level.